source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp

```diff
@@ -106,9 +106,22 @@ namespace RendererQml
         AddContainerElements(bodyLayout, card->GetBody(), context);
         AddActions(bodyLayout, card->GetActions(), context);
 
+		bodyLayout->Property("onHeightChanged", Formatter() << "{generateStretchHeight(bodyLayout.children," << card->GetMinHeight() << ");}");
+
+		bodyLayout->Property("onWidthChanged", Formatter() << "{generateStretchHeight(bodyLayout.children," << card->GetMinHeight() << ")}");
+
+		bodyLayout->AddFunctions(AdaptiveCardQmlRenderer::getStretchHeight());
+
+		bodyLayout->AddFunctions(AdaptiveCardQmlRenderer::getStretchWidth());
+
+		bodyLayout->AddFunctions(AdaptiveCardQmlRenderer::getMinWidth());
+
+		bodyLayout->AddFunctions(AdaptiveCardQmlRenderer::getMinWidthActionSet());
```

Comment:
Should these function be inside `bodyLayout`? Can we have it in root itself... and then we can use `context->getCardRootId` to get the id wherever we need to call the function.

Reply:
Yes we can have the functions inside the root layout